include/aspect/particle/property/cpo.h

```diff
@@ -0,0 +1,361 @@
+           * denotes the first component of this property, all other components
+           * fill consecutive entries in the @p particle_properties vector.
+           *
+           * @param [in] position The current particle position.
+           *
+           * @param [in] solution The values of the solution variables at the
+           * current particle position.
+           *
+           * @param [in] gradients The gradients of the solution variables at
+           * the current particle position.
+           *
+           * @param [in,out] particle_properties The properties of the particle
+           * that is updated within the call of this function.
+           */
+          virtual
+          void
+          update_one_particle_property (const unsigned int data_position,
+                                        const Point<dim> &position,
+                                        const Vector<double> &solution,
+                                        const std::vector<Tensor<1,dim> > &gradients,
+                                        const ArrayView<double> &particle_properties) const;
+
+          /**
+           * This implementation tells the particle manager that
+           * we need to update particle properties every time step.
+           */
+          UpdateTimeFlags
+          need_update () const;
+
+
+          InitializationModeForLateParticles
+          late_initialization_mode () const;
+
+          /**
+           * Return which data has to be provided to update the property.
+           * The integrated strains needs the gradients of the velocity.
+           */
+          virtual
+          UpdateFlags
+          get_needed_update_flags () const;
+
+          /**
+           * Set up the information about the names and number of components
+           * this property requires.
+           *
+           * @return A vector that contains pairs of the property names and the
+           * number of components this property plugin defines.
+           */
+          virtual
+          std::vector<std::pair<std::string, unsigned int> >
+          get_property_information() const;
+
+          /**
+           * Loads particle data into variables. Intended for use by other parts of aspect.
+           */
+          static
+          void
+          load_particle_data(unsigned int cpo_index,
+                             const ArrayView<double> &data,
+                             std::vector<unsigned int> &deformation_type,
+                             std::vector<double> &volume_fraction_mineral,
+                             std::vector<std::vector<double>> &volume_fractions_mineral,
+                             std::vector<std::vector<Tensor<2,3> > > &a_cosine_matrices_mineral);
+
+
+          /**
+           * Loads particle data into variables. An extention of load_particle_data which includes the grain
+           * derivatvies. Intended mostly for use in this plugin.
```

Comment:
If this is mostly for internal use, can it be private? If not, does it make sense to not name this `_extended` but instead give it the same name as the function above, simply making it an overloaded function with more parameters?

Reply:
I can make it private, but then I can't test it with the unit test. So I went with the overloading option.